### PR TITLE
feat : Objects and Array (#39517)

### PR DESCRIPTION
**Pull request by AryanSinghalGit**

### CODE REVIEW

File: extra/utils/permissions.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    delete: [],
+    },
+    'myUsers':{
+        all: ['hod'],
+        read : ['student', 'teacher'],
+        write : ['teacher'],
+        delete: ['admin'],
+    }
+    };
+console.log(hasPermission("getUsers","trainee","read"));
```
Comment (Rahul-Successive) [inline extra/utils/permissions.js]: What's this log for?

File: extra/utils/validation.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+{
+   user.forEach(function(value,index)
+    { 
+        
+        const {traineeEmail,reviewerEmail}=user[index];
+         // console.log(traineeEmail);
+        //console.log(reviewerEmail);
+        
+        if(validateEmail(traineeEmail)){
+            true_arr.push(traineeEmail);
```
Comment (Rahul-Successive) [inline extra/utils/validation.js]: use meaningful variable names

File: extra/utils/validation.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        traineeEmail: 'Aman@successive.tech',
+        reviewerEmail: 'vinay.chaudhary@successive.tech'
+    }
+];
+let regex=/[a-z]([[-]*\w+[.]*){1,63}@successive[.]tech$/gmi;
+let email;
+//let temp=[];
+let true_arr=[];
+let false_arr=[];
+validateUser(user);
```
Comment (Rahul-Successive) [inline extra/utils/validation.js]: You have called this function here but you defined the function below, is this working and if yes, then why?

Comment (AryanSinghalGit) [reply extra/utils/validation.js]: Yes, it is working due to hoisting i.e we can use a variable before declaring.

File: extra/utils/permissions.js (hunk 1 above)
Comment (AryanSinghalGit) [reply extra/utils/permissions.js]: This is to show the returning value of function hasPermission. Should I remove this?

File: extra/utils/validation.js (hunk 2 above)
Comment (AryanSinghalGit) [reply extra/utils/validation.js]: I am fixing code by using meaningful variable names

File: extra/utils/validation.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+{
+   user.forEach(function(value,index)
+    { 
+        
+        const {traineeEmail,reviewerEmail}=user[index];
+         // console.log(traineeEmail);
+        //console.log(reviewerEmail);
+        
+        if(validateEmail(traineeEmail)){
+            validEmails.push(traineeEmail);
```
Comment (Rahul-Successive) [inline extra/utils/validation.js]: don't try to access variables which are defined outside of the function, please declare them inside

**Pull request merged**